core/src/main/scala/kafka/admin/ZkSecurityMigrator.scala

```diff
@@ -0,0 +1,48 @@
+package kafka.admin
+
+import javax.security.auth.login.LoginException
+import joptsimple.OptionParser
+import org.I0Itec.zkclient.ZkClient
+import kafka.utils.{ToolsUtils, Logging, ZKGroupTopicDirs, ZkUtils, CommandLineUtils}
+import org.apache.kafka.common.security.JaasUtils
+
+object ZkSecurityMigrator extends Logging {
+
+  def main(args: Array[String]) {
+    val parser = new OptionParser()
+
+    val jaasFileOpt = parser.accepts("jaas.file", "JAAS Config file.").
+      withRequiredArg().ofType(classOf[String])
+    val zkUrlOpt = parser.accepts("connect", "Sets the ZooKeeper connect string (ensemble). This parameter " +
+                                  "takes a comma-separated list of host:port pairs.").
+      withRequiredArg().defaultsTo("localhost:2181").ofType(classOf[String])
+    val zkSessionTimeoutOpt = parser.accepts("session.timeout", "Sets the ZooKeeper session timeout.").
+      withRequiredArg().defaultsTo("30000").ofType(classOf[Integer])
+    val zkConnectionTimeoutOpt = parser.accepts("connection.timeout", "Sets the ZooKeeper connection timeout.").
+      withRequiredArg().defaultsTo("30000").ofType(classOf[Integer])
+    val helpOpt = parser.accepts("help", "Print usage information.")
+
+    val options = parser.parse(args : _*)
+    if(options.has(helpOpt))
+      CommandLineUtils.printUsageAndDie(parser, "ZooKeeper Migration Tool Help")
+
+    if(!options.has(jaasFileOpt) ||
+      !JaasUtils.isSecure(System.getProperty(options.valueOf(jaasFileOpt)))) {
+      error("No JAAS configuration file has been found. Please make sure that "
```

Comment:
@junrao I see the problem you're pointing out now and I have fixed it and pushed a new version. I'm still working on tests for this, so there might still be some naive bugs. But let me explain what I'm thinking and the options. I need to pass parameters to connect to ZK, so I was thinking of doing in one of two ways:

1. Using system properties (-D option on the command line) for all parameters, including the JAAS file.
2- Usin JOptSimple like in all other tools I've seen.

I opted for 2, and just so that we have a uniform way of passing parameters, I used the same way for the JAAS file. But as you noticed, I still need to set the system file so that it is picked up by the ZK client. The advantage of doing it this way is that the user doesn't have to mix -D options with -- options like this:

```
./ZkSecurityMigrator -Djava.... --connect... --session.timeout
```

Let me know if you agree with this or if you have a different preference.